website_sale_show_stock_preview/static/src/js/website_sale_show_stock_preview.js

```diff
@@ -0,0 +1,49 @@
+odoo.define('website_sale_show_stock_preview.shop_stock', function (require) {
+'use strict';
+
+var publicWidget = require('web.public.widget');
+var core = require('web.core');
+var _t = core._t;
+var rpc = require('web.rpc');
+var session = require('web.session');
+var ajax = require('web.ajax');
+
+
+publicWidget.registry.WebsiteSaleShowStockPreview = publicWidget.Widget.extend({
+    selector: '#products_grid',
+    template: 'website_sale_show_stock_preview.preview_stock',
+    xmlDependencies: ['/website_sale_show_stock_preview/static/src/xml/website_sale_show_stock_preview.xml'],
+
+    start: function () {
+        var self = this;
+        var def = [this._super.apply(this, arguments)];
+        this.render_stock();
+        return def;
+    },
+    render_stock: function(){
+        var self = this;
+        var products = $(".o_wsale_product_grid_wrapper");
+        var product_ids = [];
+        _.each(products, function(elem) {
+            var image = $(elem).find('img')[0].src;
+            product_ids.push(parseInt(image.split('/')[6]));
+        });
+
+        return this._rpc({
+            route: '/sale/get_combination_info_stock_preview/',
+            params: {'product_template_ids': product_ids},
```

Comment:
Following the previous suggestion...
```suggestion
            params: {'product_template_ids': Object.keys(product_ids)},
```